packages/snippets/bin/snippets.dart

```diff
@@ -145,30 +198,38 @@ void main(List<String> argList) {
   final String libraryName = args[_kLibraryOption] as String? ?? '';
   final String elementName = args[_kElementOption] as String? ?? '';
   final String serial = args[_kSerialOption] as String? ?? '';
-  final List<String> id = <String>[];
+  late String id;
   File? output;
+  final Directory outputDirectory = filesystem.directory(args[_kOutputDirectoryOption] as String).absolute;
   if (args[_kOutputOption] != null) {
-    id.add(path.basename(
-        path.basenameWithoutExtension(args[_kOutputOption] as String)));
-    output = filesystem.file(path.absolute(args[_kOutputOption] as String));
+    id = path.basenameWithoutExtension(args[_kOutputOption] as String);
```

Comment:
Side-note: We should probably enable the `cast_nullable_to_non_nullable` lint for this repository which would require an extra null-check here.

Reply:
OK, will do. I'll do that in a follow up PR to synchronize with the main repo analysis options.

Reply:
Actually, that lint is already enabled, and I don't get any warning here, which is probably because the `operator[]` for `ArgResults` returns `dynamic`, when it probably should return `Object`. My guess is that was too breaking for `args` to do.

Reply:
Interesting. I wonder if the lint should also fire if you cast dynamic to something non-nullable...

Reply:
Filed https://github.com/dart-lang/linter/issues/2828.